experimental/dds/tree2/src/test/feature-libraries/editable-tree-2/proxies.spec.ts

```diff
@@ -555,4 +555,34 @@ describe("SharedTreeMap", () => {
 		assert.equal(root.map.has("bar"), true);
 		assert.equal(root.map.has("baz"), false);
 	});
+
+	itWithRoot("set", schema, initialTree, (root) => {
+		// Insert new value
+		root.map.set("baz", "42");
+		assert.equal(root.map.size, 3);
+		assert(root.map.has("baz"));
+		assert.equal(root.map.get("baz"), "42");
+
+		// Override existing value
+		root.map.set("baz", "37");
+		assert.equal(root.map.size, 3);
+		assert(root.map.has("baz"));
+		assert.equal(root.map.get("baz"), "37");
+
+		// "Un-set" existing value
+		root.map.set("baz", undefined);
+		assert.equal(root.map.size, 2);
+		assert(!root.map.has("baz"));
+	});
+
+	itWithRoot("delete", schema, initialTree, (root) => {
```

Comment:
We need to update the `SharedTreeMap` type, too. It is currently typed as a `Map`. I think it should be typed as a `ReadonlyMap & { set: ..., delete: ... }`. The same pattern as SharedTreeList, in types.ts.

Otherwise, root.map will have a `delete` with a boolean return value, and a `clear` method. It does here, right?

Reply:
Ah, good call. You're right. Will update.

Reply:
Since `clear` is the only operation we aren't supporting currently, and we plan to add it eventually, I went with `Omit<Map, "clear">` instead, and left a TODO comment to add clear later.

Reply:
But `Omit<Map, "clear">` still has a `delete` method that returns a boolean, and the docs that you wrote for `set` and `delete` in the editable-tree-2 API don't get surfaced to the proxy API.

Reply:
Urg, yeah you're right. I'll put up a fix for that later this morning.

Reply:
#18133